ui/selectors/assets.ts

```diff
@@ -40,3 +59,152 @@ export function getAssetsMetadata(state: AssetsState) {
 export function getAssetsRates(state: AssetsRatesState) {
   return state.metamask.conversionRates;
 }
+
+export const getTokenBalancesEvm = createDeepEqualSelector(
+  getSelectedAccountTokensAcrossChains,
+  getSelectedAccountNativeTokenCachedBalanceByChainId,
```

Comment:
This still seems to be tied to the selected account and ignores the address passed into the selector

Reply:
Thnx for catching that @FrederikBolding 🙏 , i refactored here [4e3e9e1](https://github.com/MetaMask/metamask-extension/pull/30389/commits/4e3e9e178eac3af1e13b97835e5e65b107654113) 🙏